### PR TITLE
Add e2e tests for vlan code path for vpc-branch-eni plugin

### PR DESCRIPTION
**Pull request by fenxiong**
*Issue #, if available:*

*Description of changes:*
Add e2e tests for vlan code path for vpc-branch-eni plugin.

### Testing
make branch-e2e-tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/vpc-branch-eni/e2eTests/vlan_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (ofiliz) [inline plugins/vpc-branch-eni/e2eTests/vlan_e2e_test.go]: Should this file be called just "e2e_test". Just for consistency with AppMesh e2e tests.

File: Makefile [hunk 2]
```diff
@@ -123,6 +123,10 @@ integration-test: $(ALL_SOURCE_FILES)
 e2e-test:  $(ALL_SOURCE_FILES) all-binaries
 	sudo -E CNI_PATH=$(CUR_DIR)/$(BUILD_DIR) go test -v -tags e2e_test -race -timeout 120s ./...
 
+.PHONY: branch-e2e-tests
```
[the remark this answers is not in the document]
Comment (fenxiong) [reply Makefile]: updated target name to vpc-branch-eni-e2e-tests and update branch build tag to vpc_branch_eni.

File: plugins/vpc-branch-eni/e2eTests/vlan_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+func validateAfterDel(t *testing.T) {
+	// Check branch link is deleted.
+	_, err := netlink.LinkByName(ifName)
+	assert.Error(t, err)
+}
+
+// getEnvOrDefault gets the value of an env var. It returns the fallback value
+// if the env var is not set.
+func getEnvOrDefault(name string, fallback string) string {
```
[the remark this answers is not in the document]
Comment (fenxiong) [reply plugins/vpc-branch-eni/e2eTests/vlan_e2e_test.go]: was copied from elsewhere. fixed.

File: plugins/vpc-branch-eni/e2eTests/vlan_e2e_test.go (hunk 1 above)
Comment (fenxiong) [reply plugins/vpc-branch-eni/e2eTests/vlan_e2e_test.go]: updated

**Pull request merged**